packages/babel-plugin-transform-async-to-promises/src/index.js

```diff
@@ -0,0 +1,112 @@
+/* Babel 7 transform */
+
+const babylon = require("babylon");
+const generate = require("@babel/generator").default;
+
+function printNode(ast) {
+  return generate(ast, {}).code;
+}
+
+function parse(code) {
+  return babylon.parse(code, {
+    allowImportExportEverywhere: true,
+    allowReturnOutsideFunction: true,
+    allowSuperOutsideMethod: true,
+  }).program;
+}
+
+const transform = require("nodent-transform").transform;
+
+function transformAsyncToPromises(api, options) {
+  let requiresTranspilation;
+  const runtime = options.runtime ? options.runtime : null;
+  let opts = {
+          // Code generation options
+          es6target: false,
+          babelTree: true,
+          engine: false,
+          generators: false,
+          promises: true,
+          lazyThenables: false,
+          wrapAwait: true,
+          noRuntime: !runtime,
+          $runtime: runtime,
+          generatedSymbolPrefix: "$",
+          $return: "$return",
+          $error: "$error",
+          $arguments: "$args",
+          $Promise: "Promise",
+          $asyncspawn: "$asyncspawn",
+          $asyncbind: "$asyncbind",
+          $makeThenable: "$makeThenable",
+        }; 
+
+  if (options.codeGenerationOptions) {
+      let keys = Object.keys(options.codeGenerationOptions);
+      for (let i=0; i<keys.length;i++) {
```

Comment:
per previous comments, this can just be replaced with `Object.assign(opts, options.codeGenerationOptions);`

Reply:
(IIRC we use transform-object-rest-spread in babel itself so) alternatively,
```ts
const opts = {
  // Code generation options
  es6target: false,
  // ...
  $makeThenable: "$makeThenable",
  ...options.codeGenerationOptions
};
```

Reply:
If you look at the code, there's a conditional to prevent unknown keys being placed in the destination object. This is not the same as replacing it with `Object.assign`, which unconditionally copies all keys, irrespective of their existence in the destination.

I'm not familiar enough with the semantics of the Object spread operator to know if it is a suitable replacement for this simple loop. Is there something about `for` loops you don't like?

Reply:
Just the unreadability, hence this very confusion :-)